### PR TITLE
Zcmp test update

### PR DESCRIPTION
**Pull request by silabs-mateilga**
Updates of testes to fill coverage holes with ZCMP instructions:
pushpop_debug_triggers: added tests that verify triggers on first and last address of the sequence
debug_test_2: added tests that verify push/pop instructions vs haltreq arriving mid sequence
pma: added tests verifying push/pop instructions that cross main/IO border mid sequence

link2cov updated for all additions.

ci_check passes

### CODE REVIEW

File: cv32e40s/docs/VerifPlans/Simulation/Zc/RV32Zc_Extension_Instructions.json
Comment (silabs-robin) [inline cv32e40s/docs/VerifPlans/Simulation/Zc/RV32Zc_Extension_Instructions.json:1]: > link2cov updated for all additions

Nice one. I had forgotten about this best practice.

File: cv32e40s/tests/programs/custom/pushpop_debug_triggers/pushpop_debug_triggers.c [hunk 1]
```diff
@@ -293,7 +295,13 @@ static void  test_pop_debug_trigger(void){
 }
 
 int main(int argc, char **argv){
-  let_dmode_setup_triggers();
+  let_dmode_setup_triggers(2);
+  test_push_debug_trigger();
+  test_pop_debug_trigger();
+  let_dmode_setup_triggers(1); // trigger at first address
```
Comment (silabs-robin) [inline cv32e40s/tests/programs/custom/pushpop_debug_triggers/pushpop_debug_triggers.c:301]: Should this be index 0?

File: cv32e40s/tests/programs/custom/pushpop_debug_triggers/pushpop_debug_triggers.c [hunk 2]
```diff
@@ -293,7 +295,13 @@ static void  test_pop_debug_trigger(void){
...
+  test_push_debug_trigger();
+  test_pop_debug_trigger();
+  let_dmode_setup_triggers(3); // trigger at last address //TODO find address
```
Comment (silabs-robin) [inline cv32e40s/tests/programs/custom/pushpop_debug_triggers/pushpop_debug_triggers.c:304]: There is a todo left.

**Review by silabs-robin — COMMENTED** [review]
I've looked at the json files and the pushpop test. (Not "pma" or "debug_test2".)

**Review by silabs-robin — COMMENTED** [review]
Some more comments:

File: cv32e40s/tests/programs/custom/pushpop_debug_triggers/pushpop_debug_triggers.c (hunk 1 above)
Comment (silabs-mateilga) [reply cv32e40s/tests/programs/custom/pushpop_debug_triggers/pushpop_debug_triggers.c:301]: Visual inspection of the waveform shows that index 1 is correct.

File: cv32e40s/tests/programs/custom/debug_test2/debug_test2.c [hunk 3]
```diff
@@ -1134,6 +1140,113 @@ uint32_t request_ebreak_3x(uint32_t index, uint8_t report_name) {
...
+
+      # Push to temporary "sp"
+      cm.push {x1, x8-x9, x18-x27}, -64
+
+      # Restore old "sp"
+      mv sp, t0
+    )"::: "t0"
+  );
+
+  test_fail += *g_debug_status > 1 ? 1 : 0;
```
[the remark this answers is not in the document]
Comment (silabs-mateilga) [reply cv32e40s/tests/programs/custom/debug_test2/debug_test2.c:1183]: It is set to 1 at line 1163, and is cleared by debug handler.

File: cv32e40s/tests/programs/custom/debug_test2/debug_test2.c [hunk 4]
```diff
@@ -1134,6 +1140,113 @@ uint32_t request_ebreak_3x(uint32_t index, uint8_t report_name) {
...
+    return 0;
+  }
+
+  debug_req_ctrl = (debug_req_control_t) {
+    .fields.value            = 1,
+    .fields.pulse_mode       = 1,
+    .fields.rand_pulse_width = 0,
+    .fields.pulse_width      = 0x1fff,
+    .fields.rand_start_delay = 0,
+    .fields.start_delay      = 15
```
[the remark this answers is not in the document]
Comment (silabs-mateilga) [reply cv32e40s/tests/programs/custom/debug_test2/debug_test2.c:1160]: I have verified this visually by waveform, and we rely on ISS for verifying. I'm open to suggestions for better proof here.

File: cv32e40s/tests/programs/custom/pushpop_debug_triggers/pushpop_debug_triggers.c [hunk 5]
```diff
@@ -122,7 +123,7 @@ static void  setup_triggers(void){
   mcontrol6.fields.match = 0;  // (match exact address)
   mcontrol6.fields.type  = 6;
 
-  trigger_addr = (uint32_t) &(g_pushpop_area[2]);  // (arbitrary index)
+  trigger_addr = (uint32_t) &(g_pushpop_area[index]);  // (arbitrary index)
```
Comment (silabs-robin) [inline cv32e40s/tests/programs/custom/pushpop_debug_triggers/pushpop_debug_triggers.c:126]: `// (arbitrary index)`

This is no longer true.

But I don't really mind leaving it there, unless you already are about to change something else in the source.

File: cv32e40s/tests/programs/custom/pushpop_debug_triggers/pushpop_debug_triggers.c (hunk 1 above)
Comment (silabs-mateilga) [reply cv32e40s/tests/programs/custom/pushpop_debug_triggers/pushpop_debug_triggers.c:301]: I'm not sure, but it lines up in simulation. We could probably double check this.

File: cv32e40s/tests/programs/custom/debug_test2/debug_test2.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (silabs-mateilga) [reply cv32e40s/tests/programs/custom/debug_test2/debug_test2.c]: Ah, this stems from trying to be quick with other people's code, I had misunderstood the test reporting structure of this test. I've fixed it in the latest commit.

**Comment by silabs-mateilga** [comment]
All issues should be covered now, and I've re-ran ci_check.

**Review by silabs-robin — APPROVED** [review]
> All issues should be covered now, and I've re-ran ci_check.

Looking real good.

Approving based on:
Test status ok.
All feedback conversations resolved.
Good addition.

**Pull request merged**